src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -81,7 +84,19 @@ internal object SyncRoot
             return codeBase;
         }
 
-        public override string? CodeBase => GetCodeBase(false);
+        public override string? CodeBase
+        {
+            get
+            {
+                var runtimeAssembly = this;
+                if (IsInBundle(new QCallAssembly(ref runtimeAssembly)))
+                {
+                    // Not supported if the assembly was loaded from memory
+                    throw new PlatformNotSupportedException(SR.PlatformNotSupported_CodeBase);
```

Comment:
I think `PlatformNotSupportedException` is not the best exception to throw here. I understand that .net native used this exception, and we may keep using it for that reason. Unlike .net native, it is difficult to conceptualize .net apps and .net single-file apps to be separate platforms. Probably a better choice would be `InvalidOperationException`.

Also, should we tag this PR as a breaking change, because `Assembly.CodeBase` couldn't throw before, and now it does.